src/strategies_and_params/replicastrategy.jl

```diff
@@ -49,58 +49,95 @@ NoStats(N=1) = NoStats{N}()
+``` 
+is the (right) eigenvector of ``\\hat{G}``.
+
+For a k-tuple of input operators `(A_1,..., A_k)`, overlaps of ``f^{-1} A_k f^{-1}`` are reported 
+as `c{i}_Op{k}_c{j}`. Overlaps of ``f^{-2}`` are reported *last* as `c{i}_Op{k+1}_c{j}`.
+
+In either case, the untransformed vector-vector overlap `c{i}_dot_c{j}` 
+can be omitted with the flag `vecnorm=false`.
 """
-struct AllOverlaps{N,M,O<:NTuple{M,AbstractHamiltonian}} <: ReplicaStrategy{N}
+struct AllOverlaps{N,M,O<:NTuple{M,AbstractHamiltonian},B} <: ReplicaStrategy{N}
     operators::O
 end
 
-function AllOverlaps(num_replicas=2, operator=nothing)
+function AllOverlaps(num_replicas=2; operator=nothing, transform=nothing, vecnorm=true)
     if isnothing(operator)
         operators = ()
     elseif operator isa Tuple
         operators = operator
     else
         operators = (operator,)
     end
-    return AllOverlaps{num_replicas,length(operators),typeof(operators)}(operators)
+    if isnothing(transform)
+        ops = operators
+    else
+        fsq = Rimu.Hamiltonians.TransformUndoer(transform)
+        ops = (map(op -> Rimu.Hamiltonians.TransformUndoer(transform, op), operators)..., fsq)
+    end    
+    if !vecnorm && length(ops) == 0
+        return NoStats(num_replicas)
+    end
+    return AllOverlaps{num_replicas,length(ops),typeof(ops),vecnorm}(ops)
 end
+@deprecate AllOverlaps(num_replicas, operator) AllOverlaps(num_replicas; operator)
 
-function replica_stats(rs::AllOverlaps, replicas::NTuple{N}) where {N}
+function replica_stats(rs::AllOverlaps{N,<:Any,<:Any,B}, replicas::NTuple{N}) where {N,B}
     # Not using broadcasting because it wasn't inferred properly.
     vecs = ntuple(i -> replicas[i].v, Val(N))
-    return all_overlaps(rs.operators, vecs)
+    return all_overlaps(rs.operators, vecs, B)
 end
 
 """
-    all_overlaps(operators, vectors)
+    all_overlaps(operators, vectors, vecnorm=true)
 
 Get all overlaps between vectors and operators. This function is overloaded for `MPIData`.
+The flag `vecnorm` can disable the vector-vector overlap `c{i}_dot_c{j}`.
 """
-function all_overlaps(operators::Tuple, vecs::NTuple{N,AbstractDVec}) where {N}
+function all_overlaps(operators::Tuple, vecs::NTuple{N,AbstractDVec}, vecnorm=true) where {N}
     T = promote_type((valtype(v) for v in vecs)..., eltype.(operators)...)
     names = String[]
     values = T[]
     for i in 1:N, j in i+1:N
-        push!(names, "c$(i)_dot_c$(j)")
-        push!(values, dot(vecs[i], vecs[j]))
+        if vecnorm
+            push!(names, "c$(i)_dot_c$(j)")
+            push!(values, dot(vecs[i], vecs[j]))
+        end
         for (k, op) in enumerate(operators)
             push!(names, "c$(i)_Op$(k)_c$(j)")
             push!(values, dot(vecs[i], op, vecs[j]))
         end
     end
-
-    num_reports = (N * (N - 1) ÷ 2) * (length(operators) + 1)
+    
+    reports_per_replica = vecnorm ? length(operators) + 1 : length(operators)
+    num_reports = (N * (N - 1) ÷ 2) * reports_per_replica 
```

Comment:
```suggestion
    num_reports = (N * (N - 1) ÷ 2) * ( length(operators) + VECNORM )
```
`VECNORM` is a boolean here and adds as 0 or 1. It should now be type information, as is `N` and `length(operators)`. Thus the compiler should be able to infer `num_reports`, which is vital information for determining the return type.

I suggest that you first check whether you have a type instability with `@code_warntype` or with `@inferred` before changing anything. Then you can try my suggestions. I hope it will help.

Reply:
yes these suggestions worked. no type instability now